models.go

```diff
@@ -89,17 +89,25 @@ type IssuerResponse struct {
 	TokensNotBefore *int    `json:"tokens-not-before,omitempty"`
 }
 
+// ResourcePermission represents a permission granted to a resource
+type ResourcePermission struct {
+	RSID           *string  `json:"rsid"`
+	RSName         *string  `json:"rsname,omitempty"`
+	Scopes         []string `json:"scopes,omitempty"`
+	ResourceScopes []string `json:"resource_scopes,omitempty"`
+}
+
 // RetrospecTokenResult is returned when a token was checked
 type RetrospecTokenResult struct {
-	Permissions map[string]string `json:"permissions,omitempty"`
-	Exp         *int              `json:"exp,omitempty"`
-	Nbf         *int              `json:"nbf,omitempty"`
-	Iat         *int              `json:"iat,omitempty"`
-	Aud         *StringOrArray    `json:"aud,omitempty"`
-	Active      *bool             `json:"active"`
-	AuthTime    *int              `json:"auth_time,omitempty"`
-	Jti         *string           `json:"jti,omitempty"`
-	Type        *string           `json:"typ,omitempty"`
+	Permissions []*ResourcePermission `json:"permissions,omitempty"`
```

Comment:
we don't have the unit test to check permissions, could you implement it?
the test `TestGocloak_RequestPermission` just validates that the Keycloak doesn't return an error

Reply:
Sure, once we figure out the best way to represent permissions.

Reply:
I don't think the `RequestPermission` method works as written. To request permissions you need to pass a bearer token. The way it's implemented it's basically the same as the `Login` method. It would need to be changed to take a token, and slice of permissions as well as changing the grant type to `urn:ietf:params:oauth:grant-type:uma-ticket`. As shown here:
https://www.keycloak.org/docs/latest/authorization_services/#_service_obtaining_permissions

I'm not sure that falls within this pull request though.

Reply:
I think we just didn't use the request permissions function, nobody ever really test it.
So lets merge this PR. And you you have a time and would like to implement the proper test then welcome :)